R/utils.R

```diff
@@ -68,3 +68,17 @@ compareReal <- function(actual, expected, tol) {
   return( finiteValuesCloseEnough && nonFiniteValuesIdentical )
 }
 
+#' Throw assert error used when writing custom assertions
```

Comment:
What do you mean custom assertions? Why not focus no building the api around custom matchers as we have already done? They only need to return TRUE or FALSE

Reply:
I added such function to make writing custom assertions easier. Have you read this part of the vignette (https://github.com/bedatadriven/hamcrest/commit/81ec331c0f4bf84d2e5f3ff2aca167db75ebee00)?

I agree that the name is not so great and we can change it.